sw/device/lib/dif/dif_hmac.h

```diff
@@ -81,28 +43,16 @@ typedef enum dif_hmac_endianness {
   kDifHmacEndiannessLittle,
 } dif_hmac_endianness_t;
 
-/**
- * Error codes for HMAC functions that may exhibit generic failures.
- */
-typedef enum dif_hmac_result {
-  /** No error occurred. */
-  kDifHmacOk = 0,
-  /** An unknown error occurred. */
-  kDifHmacError = 1,
-  /** An invalid argument was provided. */
-  kDifHmacBadArg = 2,
-} dif_hmac_result_t;
-
 /**
  * Error codes for HMAC FIFO operations that may fail.
  */
 typedef enum dif_hmac_fifo_result {
```

Comment:
nit: These specialized result types seem less ergonomic now that there is a global error space.

Reply:
agreed, do you think it would be best to just stick with the four basic error types we have in dif_base.h, i.e. kDifOk, kDifError, kDifBadArg, and kDifLocked? or expand this set of global errors to include IP specific results as well? I think I'm leaning towards the former in the name of simplicity but I may not be considering all angles.

Reply:
Fifo, digest, etc. are very common design features - they could very well be added to the global error space.

Reply:
Sure thing! I will add them there, and any other common design features.

Reply:
Fixed.